dll/win32/shimgvw/util.c

```diff
@@ -131,28 +130,119 @@ DoShellContextMenu(HWND hwnd, IContextMenu *pCM, PCWSTR File, LPARAM lParam)
     g_pContextMenu = NULL;
 }
 
-void
-DoShellContextMenuOnFile(HWND hwnd, PCWSTR File, LPARAM lParam)
+HRESULT
+GetUIObjectOfPath(HWND hwnd, PCWSTR File, REFIID riid, void **ppv)
 {
     HRESULT hr;
     IShellFolder *pSF;
     PCUITEMID_CHILD pidlItem;
     PIDLIST_ABSOLUTE pidl = ILCreateFromPath(File);
     if (pidl && SUCCEEDED(SHBindToParent(pidl, IID_PPV_ARG(IShellFolder, &pSF), &pidlItem)))
     {
-        IContextMenu *pCM;
-        hr = IShellFolder_GetUIObjectOf(pSF, hwnd, 1, &pidlItem, &IID_IContextMenu, NULL, (void**)&pCM);
+        hr = IShellFolder_GetUIObjectOf(pSF, hwnd, 1, &pidlItem, riid, NULL, ppv);
+        IShellFolder_Release(pSF);
+    }
+    SHFree(pidl);
+    return hr;
+}
+
+void
+DoShellContextMenuOnFile(HWND hwnd, PCWSTR File, LPARAM lParam)
+{
+    IContextMenu *pCM;
+    HRESULT hr = GetUIObjectOfPath(hwnd, File, IID_PPV_ARG(IContextMenu, &pCM));
+    if (SUCCEEDED(hr))
+    {
+        DoShellContextMenu(hwnd, pCM, File, lParam);
+        IContextMenu_Release(pCM);
+    }
+}
+
+typedef struct
```

Comment:
```suggestion
typedef struct _ENABLECOMMANDDATA
```
to avoid anonymous structs in the debugger.